test/test_fx.py

```diff
@@ -714,6 +715,17 @@ def test_wrong_topo(self):
         nodes[2], nodes[3] = nodes[3], nodes[2]
         with self.assertRaisesRegex(RuntimeError, 'was used before it has been defined'):
             graph.lint()
+    def test_find_single_partition(self):
```

Comment:
This test is a good start, it verifies the partition is applied correctly, but it should also validate the DAG is correctly generated as well.